compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -52,6 +60,11 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     else nullables.toList
   }
 
+  private def needsBoxing(tp: Type)(using Context): Boolean = tp != NoType && tp != defn.UnitType && tp.classSymbol.isPrimitiveValueClass
```

Comment:
```suggestion
  private def needsBoxing(tp: Type)(using Context): Boolean = tp != NoType && tp.classSymbol.isPrimitiveValueClass
```
`Unit` is covered in `isPrimitiveValueClass`.